lib/cdo/shared_constants.rb

```diff
@@ -670,16 +670,12 @@ module SharedConstants
   AI_TUTOR_CHAT_MODEL_VERISON = 'gpt-3.5-turbo-1106'
 
   # These reflect the 'status' of an AI Tutor Interaction
-  AI_TUTOR_INTERACTION_SAVE_STATUS = {
+  AI_TUTOR_INTERACTION_STATUS = {
     ERROR: 'error',
-    PERSONAL: 'personal',
-    PROFANITY: 'profanity',
-    INAPPROPRIATE: 'inappropriate',
+    PII_VIOLATION: 'pii_violation',
+    PROFANITY_VIOLATION: 'profanity_violation',
```

Comment:
Changes in these two constants + rationale:
1. Rename `AI_TUTOR_INTERACTION_SAVE_STATUS` to `AI_TUTOR_INTERACTION_STATUS` because, in this case, "saved" and "ok" are the same, so the original name included the success case described within it.
2. Rename "personal" to "PII_VIOLATION" and "profanity" to "PROFANITY_VIOLATION" to match the AI TA code: https://github.com/code-dot-org/code-dot-org/blob/d779649f96aa655fd5702215e40a66ae5910e782/lib/cdo/shared_constants.rb#L62. I think this also hints that these two types of messages map directly to checks we already have in our system (WebPurify checking for PII and profanity), instead of generically being personal, for example.
3. Move "email", "address", and "phone" into their own constant. These are not interaction statuses, they're types of PII that might be flagged, so I think they should live somewhere else. 

Doing this refactor also allowed me to build my initial status filter to map directly to the states of this "AI_TUTOR_INTERACTION_STATUS" constant without unused fields.

Reply:
Makes sense to me!

Reply:
This seems reasonable to me! did you find any usages of the email/phone/address values in Ruby? It looks like they might have been moved there in an earlier refactor, might make sense for them just to live in a JS file:

https://github.com/code-dot-org/code-dot-org/pull/56361/files

Reply:
I expected to, but I didn't, Ben! Some of my tasks include sending AI Tutor chats through the PII/profanity filters and ensuring they're flagged appropriately. Do you mind leaving this here until I decide whether to use them on the backend? I filed a follow-up task to remove if I don't end up using them: https://codedotorg.atlassian.net/browse/CT-470